web-registry/src/lib/wallet/hooks/useOnAccountChange.tsx

```diff
@@ -1,23 +1,91 @@
-import { useEffect } from 'react';
+import { MutableRefObject, useEffect, useState } from 'react';
+import {
+  ApolloClient,
+  NormalizedCacheObject,
+  useApolloClient,
+} from '@apollo/client';
+import { useQuery } from '@tanstack/react-query';
+import WalletConnect from '@walletconnect/client';
+import { useSetAtom } from 'jotai';
 
+import { UseStateSetter } from 'types/react/use-state';
+import {
+  addWalletModalConnectAtom,
+  addWalletModalRemoveAtom,
+} from 'lib/atoms/modals.atoms';
+import { getPartyByAddrQuery } from 'lib/queries/react-query/registry-server/graphql/getPartyByAddrQuery/getPartyByAddrQuery';
+
+import { usePartyInfos } from 'pages/ProfileEdit/hooks/usePartyInfos';
+import { DEFAULT_NAME } from 'pages/ProfileEdit/ProfileEdit.constants';
+
+import { chainInfo } from '../chainInfo/chainInfo';
 import { Wallet } from '../wallet';
-import { WalletType } from '../walletsConfig/walletsConfig.types';
+import { getWallet } from '../wallet.utils';
+import { WalletConfig, WalletType } from '../walletsConfig/walletsConfig.types';
+import { AddAddressParams } from './useAddAddress';
 import { ConnectWalletType } from './useConnectWallet';
 
 type Props = {
   wallet: Wallet;
   connectWallet: ConnectWalletType;
+  keplrMobileWeb: boolean;
+  walletConfigRef: MutableRefObject<WalletConfig | undefined>;
+  walletConnect?: WalletConnect;
+  accountId?: string;
+  addAddress: (params: AddAddressParams) => Promise<void>;
+  setAccountChanging: UseStateSetter<boolean>;
 };
 
-export const useOnAccountChange = ({ wallet, connectWallet }: Props): void => {
+export const useOnAccountChange = ({
+  wallet,
+  connectWallet,
+  keplrMobileWeb,
+  walletConfigRef,
+  walletConnect,
+  accountId,
+  addAddress,
+  setAccountChanging,
+}: Props): void => {
+  const [newWallet, setNewWallet] = useState<Wallet | undefined>();
+  const graphqlClient =
+    useApolloClient() as ApolloClient<NormalizedCacheObject>;
+  const { data: partyByAddr, isFetching } = useQuery(
+    getPartyByAddrQuery({
+      client: graphqlClient,
+      addr: newWallet?.address ?? '',
+      enabled: !!newWallet?.address && !!graphqlClient,
+    }),
+  );
+  const { party, defaultAvatar } = usePartyInfos({ partyByAddr });
+
+  const setAddWalletModalConnectAtom = useSetAtom(addWalletModalConnectAtom);
+  const setAddWalletModalRemoveAtom = useSetAtom(addWalletModalRemoveAtom);
+
   useEffect(() => {
     if (typeof window === 'undefined') {
       return;
     }
 
-    const listener = (): void => {
+    const listener = async (): Promise<void> => {
```

Comment:
A small comment at the beginning of the hook might be helpful to get a quick understanding of the effect is doing.
Could something like
`// Update wallet or directly connect it for Keplr mobile browser`